File: app/src/main/java/com/example/flashcardapp/FlashcardTopicCrossRef.java
// File: FlashcardTopicCrossRef.java
package com.example.flashcardapp;

public class FlashcardTopicCrossRef {
    private int flashcardId;
    private int topicId;

    // Constructors
    public FlashcardTopicCrossRef() {}

    public FlashcardTopicCrossRef(int flashcardId, int topicId) {
        this.flashcardId = flashcardId;
        this.topicId = topicId;
    }

    // Getters and Setters
    public int getFlashcardId() { return flashcardId; }
    public void setFlashcardId(int flashcardId) { this.flashcardId = flashcardId; }

    public int getTopicId() { return topicId; }
    public void setTopicId(int topicId) { this.topicId = topicId; }
}

File: app/src/main/java/com/example/flashcardapp/data/FlashcardTopicCrossRef.kt
package com.example.flashcardapp.data

import androidx.room.Entity

@Entity(tableName = "flashcard_topic_cross_ref", primaryKeys = ["flashcardId", "topicId"])
 data class FlashcardTopicCrossRef(
    val flashcardId: Int,
    val topicId: Int
)
